rspotify-model/src/auth.rs

```diff
@@ -95,6 +93,109 @@ impl Token {
     }
 }
 
+/// An error reading a cached [`Token`].
+#[derive(Debug, Error)]
+#[error("failed to read token from cache")]
+pub enum ReadTokenCacheError {
+    /// There was an error reading the cache file into memory.
+    Reading(#[source] ReadFileError),
+
+    /// There was an error deserializing the contents of the cache file.
+    Deserializing(#[source] serde_json::Error),
+}
+
+/// An error writing a [`Token`] to its cache.
+#[derive(Debug, Error)]
+#[non_exhaustive]
+#[error("failed to write token to cache")]
+pub struct WriteTokenCacheError {
+    /// The underlying error in writing the [`Token`] file cache.
+    #[source]
+    pub inner: WriteFileError,
+}
+
+fn read_file(path: &Path) -> Result<Vec<u8>, ReadFileError> {
+    fs::read(path).map_err(|inner| ReadFileError {
+        inner,
+        path: Box::from(path),
+    })
+}
+
+/// An error reading a file.
+#[derive(Debug, Error)]
+#[error("failed to read file {}", path.display())]
+pub struct ReadFileError {
+    // Intentionally not exposed to allow future API evolution, e.g. moving this to a enum variants
```

Comment:
Nitpick: there should be a comma after `e.g.`. I think this also breaks the 80-character rule.

Reply:
Oh huh, that’s an American thing.

Reply:
Is it? I might be wrong, though. I've looked into it when I was writing papers or similars, and it makes sense to me, since you could replace `e.g.` with `for example`, which is usually followed by a comma (https://english.stackexchange.com/questions/16172/should-i-always-use-a-comma-after-e-g-or-i-e).

Reply:
Ah yeah, seems like it's more British:

> So, in short, I am happy to say that, as a Brit, I am not wrong in my refusal to add the comma, but then again, neither are the subsequent editors of the text. It just depends upon which flavour of English one uses natively. ;-)

Not like it matters for this library anyway. I'm neither British nor American, so my own English is just a weird mix of both.